src/sentry/taskworker/registry.py

```diff
@@ -0,0 +1,151 @@
+        self.name = name
+        self.topic = topic
+        self.deadletter_topic = deadletter_topic
+        self.default_retry = retry
+        self.__registered_tasks = {}
+
+    @property
+    def producer(self) -> KafkaProducer:
+        if self.__producer:
+            return self.__producer
+        cluster_name = get_topic_definition(Topic.TASK_WORKER)["cluster"]
+        producer_config = get_kafka_producer_cluster_options(cluster_name)
+        self.__producer = KafkaProducer(producer_config)
+
+        return self.__producer
+
+    def get(self, name: str) -> Task:
+        if name not in self.__registered_tasks:
+            raise KeyError(f"No task registered with the name {name}. Check your imports")
+        return self.__registered_tasks[name]
+
+    def contains(self, name: str) -> bool:
+        return name in self.__registered_tasks
+
+    def register(
+        self,
+        name: str,
+        idempotent: bool | None = None,
+        deadline: timedelta | int | None = None,
+        retry: Retry | None = None,
+    ):
+        """register a task, used as a decorator"""
+
+        def wrapped(func):
+            task = Task(
+                name=name,
+                func=func,
+                namespace=self,
+                idempotent=idempotent,
+                deadline=deadline,
+                retry=retry,
+            )
+            # TODO tasks should be registered into the registry
+            # so that we can ensure task names are globally unique
+            self.__registered_tasks[name] = task
+            return task
+
+        return wrapped
+
+    def retry_task(self, taskdata: TaskActivation) -> None:
+        self.producer.produce(
+            ArroyoTopic(name=self.topic),
+            KafkaPayload(key=None, value=taskdata.SerializeToString(), headers=[]),
+        )
+
+    def send_task(self, task: Task, args, kwargs) -> None:
+        task_message = self._serialize_task_call(task, args, kwargs)
+        # TODO this could use an RPC instead of appending to the topic directly
+        # TODO callback handling
+        self.producer.produce(
+            ArroyoTopic(name=self.topic),
+            KafkaPayload(key=None, value=task_message, headers=[]),
+        )
+
+    def _serialize_task_call(self, task: Task, args: list[Any], kwargs: Mapping[Any, Any]) -> bytes:
+        retry = task.retry or self.default_retry or FALLBACK_RETRY
+
+        retry_state = RetryState(
+            attempts=retry.initial_state().attempts,
+            kind=retry.initial_state().kind,
+            discard_after_attempt=retry.initial_state().discard_after_attempt,
+            deadletter_after_attempt=retry.initial_state().deadletter_after_attempt,
+        )
+        pending_task_payload = TaskActivation(
+            id=uuid4().hex,
+            namespace=self.name,
+            taskname=task.name,
+            parameters=str(orjson.dumps({"args": args, "kwargs": kwargs})),
```

Comment:
I stringified this because `orjson.dump` creates bytes and `TaskActivation` is expecting the parameters to be string

Reply:
Should this use decode?

Reply:
str is almost never the right thing -- this actually makes a nonsense `"b'{"json": "string"}'"` as written -- you want `.decode()` to convert from bytes to str (usually just `bytes_object.decode()` which uses utf-8 by default)

Reply:
Updated to use decode() now.